src/main/kotlin/com/mparticle/kits/AppboyKit.kt

```diff
@@ -216,6 +254,42 @@ open class AppboyKit : KitIntegration(), AttributeListener, CommerceListener,
 
     override fun logEvent(event: CommerceEvent): List<ReportingMessage> {
         val messages: MutableList<ReportingMessage> = LinkedList()
+
+        //For CommerceEvent, Event Name is not required to generate hash. So, it will be always null.
+        event.products?.forEach {
+            it.customAttributes?.let {
+                changeUserArray(it, CommerceEventUtils.getEventType(event), null, true)
+            }
+        }
+        event.transactionAttributes?.let {
+            val map = mapOf(
+                "Affiliation" to it.affiliation,
+                "Revenue" to it.revenue?.toString(),
+                "Shipping" to it.shipping?.toString(),
+                "Tax" to it.tax?.toString(),
+                "CouponCode" to it.couponCode?.toString(),
+                "Id" to it.id.toString()
+            ).mapValues { it.value.toString() }
+            changeUserArray(map, CommerceEventUtils.getEventType(event), null, true)
+        }
+        event.customAttributeStrings?.let {
+            changeUserArray(it, CommerceEventUtils.getEventType(event), null, true)
+        }
+        event.impressions?.forEach { impression ->
+            impression.products.forEach { product ->
+                product.customAttributes?.let {
```

Comment:
same comment as above - have you tested if the UI even populates a custom attributes that's sent into the product within a impression? if so, I doubt the hash here will be right...?